Sources/StreamChat/Extensions/BidirectionalCollection+Extensions.swift

```diff
@@ -47,6 +47,76 @@ extension BidirectionalCollection {
 
 extension BidirectionalCollection where Element == ChatMessage {
     func uniquelyMerged(_ insertedSortedElements: [Element]) -> [Element] {
-        uniquelyMerged(insertedSortedElements, areInIncreasingOrder: { first, second in first.createdAt < second.createdAt }, unique: { first, second in first.id != second.id })
+        uniquelyMerged(insertedSortedElements, areInIncreasingOrder: { first, second in first.createdAt < second.createdAt }, dropsExisting: { first, second in first.id == second.id })
+    }
+    
+    func uniquelyApplied(_ changes: [ListChange<Element>]) -> [Element] {
```

Comment:
It is going to need a lot of tests but currently I am planning to first create the public API layer first. No worries, a lot of tests coming after that.

In short, this is used for mutating an (native) array of channels/messages (not backed by FRC like we have in controllers where: `public var channels: LazyCachedMapCollection<ChatChannel> {`).

Reply:
We should probably review this during the next tech discussion as well. Adding it as a topic.